### PR TITLE
fork-cleaner: init at 2.3.1

### PR DESCRIPTION
**Pull request by isabelroses**
## Description of changes

Adds [caarlos0/fork-cleaner](https://github.com/caarlos0/fork-cleaner), a tool to quickly clean up unused forks on your github account.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by isabelroses** [comment]
Result of `nixpkgs-review pr 303817` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

1 package built:
- fork-cleaner

**Comment by uncenter** [comment]
Seems to be outputting the wrong version?

image

Normally I get this.

image

**Comment by isabelroses** [comment]
> Seems to be outputting the wrong version?

Resolved via a patch.

**Review by uncenter — APPROVED** [review]

File: pkgs/by-name/fo/fork-cleaner/package.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+    meta = {
+      description = "Quickly clean up unused forks on your github account";
+      homepage = "https://github.com/caarlos0/fork-cleaner";
+      changelog = "https://github.com/caarlos0/fork-cleaner/releases/tag/${version}";
+      license = lib.licenses.mit;
+      maintainers = with lib.maintainers; [isabelroses];
+      mainProgram = "fork-cleaner";
+    };
+  }
```
Comment (tomodachi94) [inline pkgs/by-name/fo/fork-cleaner/package.nix:38]: This is trivial, but the convention is to use `with lib;` for the `meta` block.
```suggestion
    meta = with lib; {
      description = "Quickly clean up unused forks on your github account";
      homepage = "https://github.com/caarlos0/fork-cleaner";
      changelog = "https://github.com/caarlos0/fork-cleaner/releases/tag/${version}";
      license = licenses.mit;
      maintainers = with maintainers; [isabelroses];
      mainProgram = "fork-cleaner";
    };
  }
```

File: pkgs/by-name/fo/fork-cleaner/package.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+    ldflags = [
+      "-s"
+      "-w"
+      "-X main.version=${version}"
+      "-X main.builtBy=nixpkgs"
+    ];
+
+    meta = {
+      description = "Quickly clean up unused forks on your github account";
```
Comment (tomodachi94) [inline pkgs/by-name/fo/fork-cleaner/package.nix:31]:
```suggestion
      description = "Quickly clean up unused forks on your GitHub account";
```

**Review by tomodachi94 — CHANGES_REQUESTED** [review]
Lgtm, just some tangential things.

File: pkgs/by-name/fo/fork-cleaner/package.nix (hunk 1 above)
Comment (uncenter) [reply pkgs/by-name/fo/fork-cleaner/package.nix:38]: See https://github.com/NixOS/nixpkgs/pull/303997#discussion_r1564643804.

Comment (tomodachi94) [reply pkgs/by-name/fo/fork-cleaner/package.nix:38]: Ah, I see. @isabelroses: feel free to ignore this, I'll approve it regardless of what you do with this specific comment.

**Review by tomodachi94 — APPROVED** [review]
Lgtm, thanks for this PR (might use this myself).

File: pkgs/by-name/fo/fork-cleaner/package.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    src = fetchFromGitHub {
+      owner = "caarlos0";
+      repo = "fork-cleaner";
+      rev = "v${version}";
+      sha256 = "sha256-JNmpcDwmxR+s4regOWz8FAJllBNRfOCmVwkDs7tlChA=";
+    };
+
+    vendorHash = "sha256-QuIaXXkch5PCpX8P755X8j7MeNnbewWo7NB+Vue1/Pk=";
+
+    allowGoReference = true;
```
Comment (Aleksanaa) [inline pkgs/by-name/fo/fork-cleaner/package.nix:21]: Please add a comment to explain why.

**Review by Aleksanaa — COMMENTED** [review]
The format looks a bit weird, is it alejandra or something?

**Comment by isabelroses** [comment]
> The format looks a bit weird, is it alejandra or something?

Good catch ill reformat soon.

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**